### PR TITLE
feat(swarm): deprecate `ConnectionHandlerEvent::Close`

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

We've already agreed that we want to remove this. There is no reason to not let our users know about this already through a deprecation. From the deprecation alone we can see how invasive this type parameter actually is, esp. compared to how little value it adds to the library.

Related: #3591.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 0.43.7
+
+- Deprecate `ConnectionHandlerEvent::Close`.
+  See [issue 3591](https://github.com/libp2p/rust-libp2p/issues/3591) for details.
+  See [PR XXXX](https://github.com/libp2p/rust-libp2p/pull/XXXX).
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
  See [PR 4714](https://github.com/libp2p/rust-libp2p/pull/4714).
```

**Review by mxinden — APPROVED** [review]

**Comment by thomaseizinger** [comment]
@mxinden What is the procedure again for getting this into `master`? I've released the tip of `v0.52` now. Do I merge `v0.52` into `master` or do I just cherry-pick the changes into a new PR towards `master`?

**Comment by mxinden** [comment]
@thomaseizinger you forgot to push the `libp2p-swarm-v0.43.7` tag.

**Comment by mxinden** [comment]
> @mxinden What is the procedure again for getting this into `master`? I've released the tip of `v0.52` now. Do I merge `v0.52` into `master` or do I just cherry-pick the changes into a new PR towards `master`?

Generally I would prefer a merge over a cherry-pick. Though in this case, given that https://github.com/libp2p/rust-libp2p/pull/4755 removes the deprecated items, I think it is fine as is.

**Comment by thomaseizinger** [comment]
> @thomaseizinger you forgot to push the `libp2p-swarm-v0.43.7` tag.

I didn't have one locally so I made one now :)

**Comment by thomaseizinger** [comment]
> > @mxinden What is the procedure again for getting this into `master`? I've released the tip of `v0.52` now. Do I merge `v0.52` into `master` or do I just cherry-pick the changes into a new PR towards `master`?
> 
> Generally I would prefer a merge over a cherry-pick. Though in this case, given that #4755 removes the deprecated items, I think it is fine as is.

I'll just manually update the changelog then.

**Comment by mxinden** [comment]
> > > @mxinden What is the procedure again for getting this into `master`? I've released the tip of `v0.52` now. Do I merge `v0.52` into `master` or do I just cherry-pick the changes into a new PR towards `master`?
> > 
> > 
> > Generally I would prefer a merge over a cherry-pick. Though in this case, given that #4755 removes the deprecated items, I think it is fine as is.
> 
> I'll just manually update the changelog then.

Whatever works best for you. I don't feel strongly here. Thank you!

**Pull request merged**